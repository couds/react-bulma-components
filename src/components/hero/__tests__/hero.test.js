import React from 'react';
import { render } from 'react-testing-library';
import { Hero } from '..';
import CONSTANTS from '../../../constants';

describe('Hero component', () => {
  it('should expose Hero head, body and footer', () => {
    expect(Hero.Head).toBeDefined();
    expect(Hero.Body).toBeDefined();
    expect(Hero.Footer).toBeDefined();
  });
  it('should have hero classname', () => {
    const { asFragment } = render(
      <Hero>
        Test <a>Give me</a>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with Bulma classname', () => {
    const { asFragment } = render(
      <Hero className="other-class this-is-a-test">
        <p>Default</p>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Hero style={{ height: 250 }}>
        <p>Default</p>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use gradient style', () => {
    const { asFragment } = render(
      <Hero color="primary" gradient>
        <p>Default</p>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render Hero with hero head, body and footer', () => {
    const { asFragment } = render(
      <Hero size="fullheight" color="primary">
        <Hero.Head renderAs="header">
          <div className="bd-notification is-info">Header</div>
        </Hero.Head>
        <Hero.Body>Body</Hero.Body>
        <Hero.Footer renderAs="footer">
          <div className="bd-notification is-danger">Footer</div>
        </Hero.Footer>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  Object.values(CONSTANTS.COLORS).map(color =>
    it(`Should use use color ${color}`, () => {
      const { asFragment } = render(
        <Hero color={color}>
          <p>Default</p>
        </Hero>
      );
      expect(asFragment()).toMatchSnapshot();
    })
  );
});
