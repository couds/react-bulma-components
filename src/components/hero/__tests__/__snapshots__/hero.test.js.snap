// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component Should use use color black 1`] = `
<DocumentFragment>
  <section
    class="hero is-black"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color danger 1`] = `
<DocumentFragment>
  <section
    class="hero is-danger"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color dark 1`] = `
<DocumentFragment>
  <section
    class="hero is-dark"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color info 1`] = `
<DocumentFragment>
  <section
    class="hero is-info"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color light 1`] = `
<DocumentFragment>
  <section
    class="hero is-light"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color link 1`] = `
<DocumentFragment>
  <section
    class="hero is-link"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color null 1`] = `
<DocumentFragment>
  <section
    class="hero"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color primary 1`] = `
<DocumentFragment>
  <section
    class="hero is-primary"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color success 1`] = `
<DocumentFragment>
  <section
    class="hero is-success"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color warning 1`] = `
<DocumentFragment>
  <section
    class="hero is-warning"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component Should use use color white 1`] = `
<DocumentFragment>
  <section
    class="hero is-white"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component should concat classname in props with Bulma classname 1`] = `
<DocumentFragment>
  <section
    class="hero other-class this-is-a-test"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component should have hero classname 1`] = `
<DocumentFragment>
  <section
    class="hero"
  >
    Test 
    <a>
      Give me
    </a>
  </section>
</DocumentFragment>
`;

exports[`Hero component should render Hero with hero head, body and footer 1`] = `
<DocumentFragment>
  <section
    class="hero is-primary is-fullheight"
  >
    <header
      class="hero-head"
    >
      <div
        class="bd-notification is-info"
      >
        Header
      </div>
    </header>
    <div
      class="hero-body"
    >
      Body
    </div>
    <footer
      class="hero-foot"
    >
      <div
        class="bd-notification is-danger"
      >
        Footer
      </div>
    </footer>
  </section>
</DocumentFragment>
`;

exports[`Hero component should use gradient style 1`] = `
<DocumentFragment>
  <section
    class="hero is-primary is-bold"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Hero component should use inline styles 1`] = `
<DocumentFragment>
  <section
    class="hero"
    style="height: 250px;"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;
